### PR TITLE
Index almost all text fields as keywords to make exact-match searches possible

### PR DESCRIPTION
**Pull request by leonardr**
This branch makes almost all of the text fields associated with a work into `filterable_text` (indexed four times including once as a keyword) rather than `basic_text` (indexed three times, no keyword). The only exception is `summary`, which is a huge chunk of text no one is going to try to match.

This will give me a testbed to improve our search techniques. We try really hard to promote close matches, but without indexing title (in particular) as a keyword, we have no way of promoting an _exact_ title match, which is a very common scenario.

Once it's all done, I hope to be able to get rid of one or two of the ways that a `filterable_text` is indexed, but I can't be sure yet.

### CODE REVIEW

File: external_search.py [hunk 1]
```diff
@@ -2660,7 +2661,7 @@ def exists(self, index, doc_type, id):
     def create_search_doc(self, query_string, filter=None, pagination=None, debug=False):
         return self.docs.values()
 
-    def query_works(self, query_string, filter, pagination, debug=False, search=None):
+    def query_works(self, query_string, filter, pagination, debug=False):
```
Comment (leonardr) [inline external_search.py]: Unrelated change -- I noticed this variable was never used.

**Review by aslagle — APPROVED** [review]
